proteus/TwoPhaseFlow/TwoPhaseFlowProblem.py

```diff
@@ -73,44 +86,106 @@ def __init__(self,
         self.rans3p_parameters = default_rans3p_parameters
         self.clsvof_parameters = default_clsvof_parameters
 
-    def assert_initialConditions(self,ns_model,nd,initialConditions):
-        assert 'pressure' in initialConditions, 'Provide pressure in ICs'
-        assert 'vel_u' in initialConditions, 'Provide vel_u in ICs'
-        assert 'vel_v' in initialConditions, 'Provide vel_v in ICs'
-        if nd==3:
-            assert 'vel_w' in initialConditions, 'Provide vel_w in ICs'
-        assert 'clsvof' in initialConditions, 'Provide clsvof in ICs'
-        if ns_model==1: #rans3p
-            assert 'pressure_increment' in initialConditions, 'Provide pressure_increment in ICs'
+        # set indice of models if ns_model and ls_model is set
+        ind = 0
+        if self.ls_model == 0:
+                self.Parameters.Models.vof.index = ind
+                ind += 1
+                self.Parameters.Models.ncls.index = ind
+                ind += 1
+                self.Parameters.Models.rdls.index = ind
+                ind += 1
+                self.Parameters.Models.ls_consrv.index = ind
+                ind += 1
+        elif self.ls_model == 1:
+            self.Parameters.Models.clsvof.index = ind
+            ind += 1
+            if self.ns_model == 0:
```

Comment:
Is there an indentation error here? I think this line and its corresponding block should happen independently of the value of `self.ls_model`.

Reply:
Actually I forgot to put the option for `ls_model == 0`, because usually we solve vof after RANS2P, and it looks like clsvof is before in your implementation, is that right? If so, we should sometimes put free surface stuff before or after RANS